src/app/qgsclipboard.cpp

```diff
@@ -131,14 +133,34 @@ void QgsClipboard::generateClipboardText( QString &textContent, QString &htmlCon
 
         for ( int idx = 0; idx < attributes.count(); ++idx )
         {
-          QString value = attributes.at( idx ).toString();
+          QString value;
+          QVariant variant = attributes.at( idx );
+          bool useJSONFromVariant = variant.canConvert( QVariant::StringList ) || variant.canConvert( QVariant::List ) || variant.canConvert( QVariant::Map );
+          useJSONFromVariant = useJSONFromVariant && ( variant.type() != QVariant::String ); // String is convertible to StringList, but we want to use toString for String variant
```

Comment:
```suggestion
          const bool useJSONFromVariant = variant.type() == QVariant::StringList || variant.type() == QVariant::List || variant.type() == QVariant::Map;
```

Basically, canConvert is a no-go now. We shouldn't use it anywhere anymore.

Reply:
ah thanks, I misunderstood your previous comment.